stl/inc/filesystem

```diff
@@ -2552,6 +2554,9 @@ namespace filesystem {
             if (_Error == __std_win_error::_Access_denied
                 && _Bitmask_includes_any(_Options_arg, directory_options::skip_permission_denied)) {
                 _Error = __std_win_error::_No_more_files;
+            } else if (_Error == __std_win_error::_File_not_found
+                       && exists(std::wstring_view(_Path.c_str(), _Path.native().length() - 1))) {

```

Comment:
After thinking about this, the `wstring_view` construction is trying really hard to avoid allocating an extra string, but we're calling `exists` which takes `const path&`, so we're going to get a string there anyways. What we're doing (via the `- 1`) is trimming away the `"*"` we just added. I'm *pretty* sure (90% sure) that we can simply capture the original `path` (i.e. that we don't need the separator that `/= L"*"sv` might have added), in which case (1) the code is simpler and (2) we get to avoid a potential allocation.